docs/adr/adr_authorization-specification.md

```diff
@@ -0,0 +1,173 @@
+
+Every ABP comprises one or more RBP, e.g. the ABP `manage_tags` stands for `tag:write`, `stock:read`, `tag_relation:read`, and `beneficiary:read`.
+
+The mapping of usergroup to ABPs, and ABP to RBPs is listed in [this document](https://docs.google.com/spreadsheets/d/1W4YWcc59wUFUWgReumdH6DQ4zU7JcTgvf6WEbdqaGHQ/edit#gid=0).
+
+### User management in Auth0
+
+[Auth0](https://auth0.com) is a service for managing user authentication and authorization. It serves as single source of truth. An authenticated user gets issued a JSON Web Token (JWT) holding their authorization information. For signing the JWT we use the RS256 algorithm: the token will be signed with our private signing key and can be verified using our public signing key.
+
+#### Reasons to use Auth0
+
+Cf. [related ADR](./docs/adr/adr_auth0.md)
+
+#### User entity in Auth0
+
+Any user registered for boxtribute has their authorization data (`app_metadata`) stored in the Auth0 database. The user attributes are:
+
+- `is_god`: `"1"` for god user, `"0"` otherwise
+- `usergroup_id`: the ID of the usergroup the user belongs to
+- `base_ids`: a list of base IDs that the user has access to
+- `organisation_id`: the ID of the organisation the user belongs to
+
+During registration, the user gets assigned a role, indicating their usergroup and the bases they belong to. The role is named like `base_1_coordinator`.
+
+From the user's role the ABPs are derived via a script (?).
+
+When the user has successfully logged in, an Auth0 post-login action script runs. The script creates a JWT with the content derived from user authorization data and their role.
+Most importantly the script creates base-specific RBPs for the current user (see below about their format).
+
+#### Specification of custom JWT
+
+The JWT contains standard and customs fields.
+
+Field name | Kind | Description | Usage
+:--- | :--- | :--- | :---
+`iss` | standard | Name of token issuer | JWT decoding
+`aud` | standard | Name of token audience | JWT decoding
+`iat` | standard | Unix timestamp of issuing datetime | -
+`exp` | standard | Unix timestamp of expiration datetime | JWT decoding
+`azp` | standard | ID of client through which the JWT was requested | traceability of application used for authentication
+`gty` | standard | Grant type | -
+`sub` | standard | User ID | see below
+`https://www.boxtribute.com/email` | custom | User email | -
+`https://www.boxtribute.com/roles` | custom | List of user's roles | -
+`https://www.boxtribute.com/base_ids` | custom | List of IDs of bases that the user has access to | see below
+`https://www.boxtribute.com/organisation_id` | custom | ID of the organisation the user belongs to | see below
+`https://www.boxtribute.com/permissions` | custom | List of RBPs that the user holds | see below
+
+### Implementation of authorization
+
+#### Dropapp
```

Comment:
@HaGuesto if relevant would be good to have your input here.

Reply:
Text I would add here:
```
When a new base is created, the following is created:
- in dropapp the usergroups: administrator (only created when an org is created), coordinator, volunteer (combination of warhoues/free shop volunteer), warehouse volunteer, free shop volunteer, label creator.
- in auth0 all roles of the section roles.
- in dropapp db table usergroups_roles a mapping between the user groups and roles.

The mapping between user groups in dropapp and roles in auth0 is needed because only one user group can be assigned to a user in dropapp, but multiple roles can be assigned to a user in auth0.

When a user is created/edited a user group must be assigned. Through the mapping the corresponding roles are then assigned in auth0 to the user.
```

Reply:
@HaGuesto is it a manual process what you describe in the first paragraph?

Reply:
> in auth0 all roles of the section roles

It's not clear to me what "section roles" mean, can you elaborate?

Reply:
I hope this makes it clearer. @pylipp